### PR TITLE
Site Settings: Fix issue with saving site settings

### LINKED ISSUE #23787: Changing site settings creates an infinite request loop
### Expected behavior
Updating a single setting should result in a single `POST` request. 

### Actual behavior
Updating a single setting results in a never-ending stream of `POST` requests. 

### Steps to reproduce the behavior
1. Set up your device so that you can monitor network traffic—e.g., send device requests through Charles Proxy. 
1. Navigate to _My Site_. 
2. Tap _More_. 
3. Scroll down and tap _Site Settings_. 
4. Scroll down to and tap _Discussion_. 
5. Toggle _Allow Comments_.

### PR DESCRIPTION
**Pull request by kean**
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/23787. The app will now use the existing method `didChangeSetting` for tracking changes in `DiscussionSettingsViewController` as a trigger for save. I also made a couple of other changes:

- Fix an issue where if you change settings fairly quickly, the app will send parallel requests
- Add error handling (was completely missing!)
- Add a spinner to the navigation bar to indicate when the save is happening (on iPad shown in the left corner)
- Remove “refresh settings” as it happens on the root screen already. There is no point to do it again here and it might mess with some of the local changes.

**Note**: ignore the first commit https://github.com/wordpress-mobile/WordPress-iOS/pull/23831/commits/b24ddf8d500b7355ff71e76298d9ed9540f621e2 as it simply fixes the formatting.

https://github.com/user-attachments/assets/8148c2f5-5ea1-4aef-939a-c66590085375

To test:

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/297">25.5</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Blog/Site Settings/DiscussionSettingsViewController.swift [hunk 1]
```diff
@@ -17,21 +19,16 @@ open class DiscussionSettingsViewController: UITableViewController {
...
         setupTableView()
-        setupNotificationListeners()
     }
 
     open override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
+
         tableView.reloadSelectedRow()
         tableView.deselectSelectedRowWithAnimation(true)
-        refreshSettings()
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Blog/Site Settings/DiscussionSettingsViewController.swift:28]: My gut feeling is this refreshing settings call is here for a reason. Is there any straightforward way to keep this call?

Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/Site Settings/DiscussionSettingsViewController.swift:28]: Yeah, it's safe to keep it. It doesn't really need to be part of this PR.
Reverting..

**Review by crazytonyli — APPROVED** [review]

**Review by dcalhoun — COMMENTED** [review]
Thanks for addressing this. Helpful for me to see the solution.

**Pull request merged**